pkg/kp/rollstore/consul_store.go

```diff
@@ -204,6 +218,15 @@ func (s consulStore) CreateRollingUpdateFromExistingRCs(u roll_fields.Update) (r
 		return roll_fields.Update{}, err
 	}
 
+	err = s.labelNewRC(u.NewRC, newRCLabels)
+	if err != nil {
+		// This is potentially bad, because the labels that were there
+		// before have been overwritten. However its unreasonable to
+		// expect that a rollback will succeed if the roll-forward
+		// didn't succeed, so just report the error
```

Comment:
That's our internal label selector, but any RC selector may be used for the RU farm. That said, it may not have that much of an impact since the RU farm would hopefully never see an RC that has lingering labels like this.